template/src/pageComponents/mint/steps/StartMintStep.tsx

```diff
@@ -1,63 +1,93 @@
 };
 
-export default function StartMintStep({ setMintStep }: StartMintProps) {
-  const { chain } = useNetwork();
+export default function StartMintStep({ setMintStep, mintStep, collectionName }: StartMintProps) {
+  const { chain } = useAccount();
   const { address } = useAccount();
 
   const contract = useCustom1155Contract();
 
   const onCorrectNetwork = chain?.id === EXPECTED_CHAIN.id;
 
-  const { config } = usePrepareContractWrite({
+  const { data: mintData } = useSimulateContract({
     address: contract.status === 'ready' ? contract.address : undefined,
     abi: contract.abi,
     functionName: 'mint',
     args: address ? [address, BigInt(1), BigInt(1), address] : undefined,
-    enabled: onCorrectNetwork,
+    query: {
+      enabled: onCorrectNetwork,
+    },
   });
 
-  // A future enhancement would be to use the `isLoading` and `isSuccess`
-  // properties returned by `useContractWrite` to indicate transaction
-  // status in the UI.
-  const { write: performMint } = useContractWrite({
-    ...config,
-    onSuccess() {
-      setMintStep(MintSteps.MINT_COMPLETE_STEP);
+  const { writeContract: performMint, error: errorMint, data: dataMint } = useWriteContract();
+
+  const { status: transactionStatus } = useWaitForTransactionReceipt({
+    hash: dataMint,
+    query: {
+      enabled: !!dataMint,
     },
-    onError(e) {
+  });
+
+  useEffect(() => {
+    if (transactionStatus === 'success') {
+      setMintStep(MintSteps.MINT_COMPLETE_STEP);
+    }
+
+    if (transactionStatus === 'error') {
       if (
-        e instanceof TransactionExecutionError &&
-        e.message.toLowerCase().includes('out of gas')
+        errorMint instanceof TransactionExecutionError &&
+        errorMint.message.toLowerCase().includes('out of gas')
       ) {
         setMintStep(MintSteps.OUT_OF_GAS_STEP);
       } else {
         setMintStep(null);
       }
-    },
-  });
+    }
+  }, [transactionStatus, setMintStep, errorMint]);
 
   const handleMint = useCallback(() => {
-    performMint?.();
-    setMintStep(MintSteps.MINT_PROCESSING_STEP);
-  }, [performMint, setMintStep]);
+    if (mintData?.request) {
+      performMint?.(mintData?.request);
+      setMintStep(MintSteps.MINT_PROCESSING_STEP);
+    }
+  }, [mintData, performMint, setMintStep]);
 
   return (
-    <Button
-      buttonContent="Mint"
-      onClick={handleMint}
-      disabled={!onCorrectNetwork}
-      className={clsx('my-4', onCorrectNetwork ? 'bg-white' : 'bg-gray-400')}
-    />
+    <>
+      {mintStep === MintSteps.START_MINT_STEP && <MintProcessingStep />}
+      {mintStep === MintSteps.OUT_OF_GAS_STEP && <OutOfGasStep setMintStep={setMintStep} />}
+      {mintStep === MintSteps.MINT_COMPLETE_STEP && (
+        <MintCompleteStep setMintStep={setMintStep} collectionName={collectionName} />
+      )}
+
+      {mintStep === null && (
+        <Button
+          buttonContent="Mint"
+          onClick={handleMint}
+          disabled={!onCorrectNetwork}
```

Comment:
sorry if this is out of scope of this simple 1:1 migration, but wouldn't it be better to disable button `status` from 3 wagmi hooks is `pending` to prevent user clicking it multiple times and sending multiple txs